thread/common/omrthread.c

```diff
@@ -2813,7 +2813,9 @@ interrupt_waiting_thread(omrthread_t self, omrthread_t threadToInterrupt)
 	 * In the 3-tier case, this prevents us from acquiring MONITOR_LOCK before the
 	 * thread to interrupt can.
 	 */
-	if (omrthread_monitor_try_enter_using_threadId(monitor, self) == 0) {
+	if ((NULL != monitor)
```

Comment:
How can threadToInterrupt->monitor be NULL?

Reply:
i was unable to figure out; more investigation is required. issue is intermittent, and it only happens with OpenLiberty after a javacore is requested using `SIGQUIT`. without the new signal changes, the issue is not observed. the new signal changes don't explicitly affect monitors. the new signal changes only change how signal handlers are invoked; functionally signal handlers are invoked similar to the old approach; the only difference may be timing due to reduced synchronization. the `sigquit` handler in OpenJ9 seems to have timing holes; a previously suppressed bug may have been triggered with the new signal changes. there may be other possibilities: something specific to OpenLiberty or a race condition within the VM during javacore generation.